spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/slf4j/Slf4jSpanStartedListener.java

```diff
@@ -0,0 +1,25 @@
+package org.springframework.cloud.sleuth.slf4j;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.slf4j.MDC;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.Trace;
+import org.springframework.cloud.sleuth.event.SpanStartedEvent;
+import org.springframework.context.ApplicationListener;
+
+/**
+ * @author Spencer Gibb
+ */
+@Slf4j
+public class Slf4jSpanStartedListener implements ApplicationListener<SpanStartedEvent> {
+
+	@Override
+	public void onApplicationEvent(SpanStartedEvent event) {
+		Span span = event.getSpan();
+		//TODO: what log level?
+		log.info("Starting span: {}", span);
+		MDC.put(Trace.SPAN_ID_NAME, span.getSpanId());
```

Comment:
And there we have MDC.put with the former 'Correlation-Id' and current 'Trace-id'.